### PR TITLE
Use libc definitions for wait module

### PR DESCRIPTION
**Pull request by Susurrus**
Not certain if these are all correct yet, but we can reuse a ton of logic from `libc` within the `wait` module.

### CODE REVIEW

File: src/sys/wait.rs [hunk 1]
```diff
@@ -112,149 +96,52 @@ impl WaitStatus {
...
-          target_os = "android"))]
 mod status {
     use sys::signal::Signal;
-    use libc::c_int;
-    use libc::SIGTRAP;
+    use libc::{self, c_int, SIGTRAP};
 
     pub fn exited(status: i32) -> bool {
-        (status & 0x7F) == 0
+        unsafe { libc::WIFEXITED(status) }
```
Comment (asomers) [inline src/sys/wait.rs]: Why `unsafe`?  `libc::WIFEXTED` and friends are not FFI functions.

Comment (Susurrus) [inline src/sys/wait.rs]: They're marked as unsafe in `libc` actually. They're actually also marked as `extern "C"` which is also interesting. I definitely don't think they need to be either of those. @alexcrichton do you know why this is?

Comment (asomers) [inline src/sys/wait.rs]: Likely just an oversight.  We could move them out of the `f!` block in libc and then they wouldn't be `unsafe` anymore.  However, I wonder if they even belong in libc at all.  There aren't many functions in libc that have more than a simple FFI call.

Comment (Susurrus) [inline src/sys/wait.rs]: These are defined as macros or functions by the spec, so it makes sense to keep them in `libc` even if they're macros on some platforms and need to be implemented.

Comment (alexcrichton) [inline src/sys/wait.rs]: Everything in libc is automatically extern and unsafe, matching the underlying implementation if it didn't use macros.

**Comment by asomers** [comment]
So according to POSIX's definition, these functions may be `unsafe`, but they happen not to be on all of nix's currently targeted platforms.  In that case, libc's decision to make them `unsafe` everywhere makes sense.

bors r+

**Comment by Susurrus** [comment]
Whoops, looking over this again, I think I actually introduced a bug in `term_signal`. We need to mask off the top bit of the status because of how `ptrace` works, and I don't think that function does it. I'll check it out tomorrow and see.

**Comment by rocallahan** [comment]
> I think I actually introduced a bug in term_signal

I don't think you did. Linux `WTERMSIG` masks off bit 7 of the status.

This PR looks good to me, though I think you could go a bit further and eliminate the `status` module, moving all its functions to the top level.

**Comment by Susurrus** [comment]
@rocallahan Thanks for checking that, yes it does look like it does the right things on Linux.

And I removed the `status` module, but made all the functions private, as I saw no reason to expose them, especially with the work you're doing in #741.

**Pull request merged**